x-pack/plugins/aiops/server/routes/queries/fetch_index_info.ts

```diff
@@ -73,20 +75,33 @@ export const fetchFieldCandidates = async (
     }
   });
 
-  const resp = await esClient.search(getRandomDocsRequest(params), {
-    signal: abortSignal,
-    maxRetries: 0,
-  });
+  // Only the deviation window will be used to identify field candidates and sample probability based on total doc count.
+  const resp = await esClient.search(
+    getRandomDocsRequest({ ...params, start: params.deviationMin, end: params.deviationMax }),
+    {
+      signal: abortSignal,
+      maxRetries: 0,
+    }
+  );
   const sampledDocs = resp.hits.hits.map((d) => d.fields ?? {});
 
   // Get all field names for each returned doc and flatten it
-  // to a list of unique field names used across all docs.
-  // and filter by list of acceptable fields and some APM specific unique fields.
+  // to a list of unique field names used across all docs
+  // and filter by list of acceptable fields.
   [...new Set(sampledDocs.map(Object.keys).flat(1))].forEach((field) => {
     if (acceptableFields.has(field)) {
       finalFieldCandidates.add(field);
     }
   });
 
-  return [...finalFieldCandidates];
+  const totalDocCount = (resp.hits.total as estypes.SearchTotalHits).value;
+
+  const minDocCount = 50000;
```

Comment:
Suggestion to rename this to MIN_DOC_COUNT if it's a static constant, unless `fetchIndexInfo` will allow `minDocCount` as an optional argument.

Reply:
Changed in [1daec7d](https://github.com/elastic/kibana/pull/143977/commits/1daec7de3c5debb92797b7c4fb65c282e70b2f54).